### ISSUE TITLE
error cb() never called!

### ISSUE THREAD

**Issue by pmanu93** [npm/npm#19644]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
"error cb() never called!" is shown when trying to execute  `npm install`..
Before that I had an annoying error related to `dezalgo` package, which I always resolve by removing  `node_modules` folder and executing `npm install`.. which sometimes results in the error in the title.
I don't know if there is some relation, but [here's the log](https://gist.github.com/pmanu93/933531af9a7081ac45ca5769c77f5871) of the `dezalgo` error.

#### How can the CLI team reproduce the problem?
I don't know, it just happens sometimes. 
Log [here](https://gist.github.com/pmanu93/0a5b1963826d623085f177f1118d9001)

### supporting information:

 - `npm -v` prints: 5.6.0
 - `node -v` prints: v8.9.4
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Linux Mint 17.3 Rosa
 - Network issues:
   - Geographic location where npm was run:
   - [x] I use a proxy to connect to the npm registry.
   - [x] I use a proxy to connect to the web.
   - [x] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by ORESoftware** [comment]
`dezalgo` is a package written by the NPM team, so this is a real concern

however, what's most likely happening is another package X is using dezalgo, and X is the one that's failing to fire a callback.

can you copy the results of this command to this conversation:

`npm la dezalgo`

that command should list all of the dependencies in your project that depend on dezalgo. run that command from the root of your project.

**Comment by ORESoftware** [comment]
fyi, `npm la` is: https://docs.npmjs.com/cli/ls

**Comment by pmanu93** [comment]
`npm la dezalgo` returns:
```sh
app@0.1.0
│ /home/pedro/dev/app
│ An Aurelia client application.
│ http://www.aurelia.io
└── (empty)
```
(probably because  I need the modules installed first?)
  

----
Other thing that I find strange is the number of processes that node creates. Is this normal?
```sh
pedro    21181     1  0 18:33 ?        00:00:00 /software/Utils/node-v8.9.4-linux-x64/bin/node /software/Utils/node-v8.9.4-linux-x64/lib/node_modules/npm/node_modules/update-notifier/check.js {"pkg":{"name":"npm","version":"5.6.0"}}
pedro    21203     1  0 18:33 ?        00:00:00 /software/Utils/node-v8.9.4-linux-x64/bin/node /software/Utils/node-v8.9.4-linux-x64/lib/node_modules/npm/node_modules/update-notifier/check.js {"pkg":{"name":"npm","version":"5.6.0"}}
pedro    21226     1  0 18:33 ?        00:00:00 /software/Utils/node-v8.9.4-linux-x64/bin/node /software/Utils/node-v8.9.4-linux-x64/lib/node_modules/npm/node_modules/update-notifier/check.js {"pkg":{"name":"npm","version":"5.6.0"}}
pedro    21249     1  0 18:33 ?        00:00:00 /software/Utils/node-v8.9.4-linux-x64/bin/node /software/Utils/node-v8.9.4-linux-x64/lib/node_modules/npm/node_modules/update-notifier/check.js {"pkg":{"name":"npm","version":"5.6.0"}}
pedro    21308     1  0 18:35 ?        00:00:00 /software/Utils/node-v8.9.4-linux-x64/bin/node /software/Utils/node-v8.9.4-linux-x64/lib/node_modules/npm/node_modules/update-notifier/check.js {"pkg":{"name":"npm","version":"5.6.0"}}
pedro    21328     1  0 18:35 ?        00:00:00 /software/Utils/node-v8.9.4-linux-x64/bin/node /software/Utils/node-v8.9.4-linux-x64/lib/node_modules/npm/node_modules/update-notifier/check.js {"pkg":{"name":"npm","version":"5.6.0"}}
pedro    21355     1  0 18:35 ?        00:00:00 /software/Utils/node-v8.9.4-linux-x64/bin/node /software/Utils/node-v8.9.4-linux-x64/lib/node_modules/npm/node_modules/update-notifier/check.js {"pkg":{"name":"npm","version":"5.6.0"}}
```

**Comment by the1mills** [comment]
Yes you need to run "npm install" before using the `npm la dezalgo` command

NPM probably uses as many processes as you have CPU cores on your machine.

**Comment by pmanu93** [comment]
The problem is.. I can't `npm install` because:
```js
npm ERR! cb() never called!
```
--- 
I already removed `.npm` folder and tried with a clean node installation... I even rebooted my pc (-.-') , but no luck.

**Comment by the1mills** [comment]
Oh yes, duh, my bad. Hmmm, this is a tough one - it's most likely a userland module problem

One thing you can try is to change the NPM version and Node version.

What versions of those are you using?

**Comment by ORESoftware** [comment]
Ok I see you are using

```
npm -v prints: 5.6.0
node -v prints: v8.9.4
```

you might have to write a script which reads the dependency tree from each package in your NPM project and see where dezalgo is being used.

I am willing to bet that NPM already has a built in tool to do this.
Use this command recursively, and simply create a list of all packages that depend on `dezalgo`:

`$ npm view X dependencies`

**Comment by ORESoftware** [comment]
in the meantime try

`npm install --production`

does that work?

**Comment by ORESoftware** [comment]
@pmanu93 I created a tool to help you in this case

try using this package

https://github.com/ORESoftware/npm-la-recovery

install it, then run it like so:

`npmlar  --local --name dezalgo`

that will give you a list of dependencies in your project that use dezalgo - it obviously works even if you cannot run `npm install`

**Comment by pmanu93** [comment]
It will have to wait until tomorrow.. unfortunately I don't have access to the PC right now :s

But thanks!

**Comment by pmanu93** [comment]
It will have to wait until tomorrow.. unfortunately I don't have access to the PC right now :s

But thanks!

**Comment by pmanu93** [comment]
@ORESoftware  So.. I tried your script today, and the result was `[]`.
Then, I tried `npm install` one more time, and this time it worked! (...go figure)
This is really annoying!!!

**Comment by pmanu93** [comment]
Here we go again...
This time I didn't remove the `node_modules` folder..
The result for `npm la dezalgo`  is [here](https://gist.github.com/pmanu93/3f14fa316fb068f53d9efd29a849ce11).. any ideia of why is this happening?

**Comment by ryanrishi** [comment]
@pmanu93 for what it's worth, I was facing the NOENT dezalgo issue too. I'm running NPM 5.6.0 and running `npm cache verify` seems to fix that part. It still comes up every few times I do `npm install`, so it's a deeper issue, but maybe `npm cache verify` will at least allow you to install dependencies.
